lib/iris/__init__.py

```diff
@@ -427,4 +428,10 @@ def load_raw(uris, constraints=None, callback=None):
 
 def sample_data_path(*path_to_join):
     """Given the sample data resource, returns the full path to the file."""
-    return os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    target = os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    if not glob.glob(target):
```

Comment:
Can we go even further. Even if it does result in files, can we prevent the following from happening:

```
sample_data_path('/absolute', 'path')
```

I'm afraid to say I have seen this in the wild, and it upset me. :cry: